service/web/server_test.go

```diff
@@ -103,3 +104,32 @@ func TestGetProjectIDFromRequest(t *testing.T) {
 		assert.Equal(t, tst.err, err)
 	}
 }
+
+func Test_getResourceNameFromRequest(t *testing.T) {
+	type test struct {
+		request interface{}
+		name    string
+		err     error
+	}
+	type testReq struct {
+	}
+	tests := []test{
+		{
+			request: &metrov1.GetSubscriptionRequest{
+				Name: "projects/project123/subscriptions/testsub",
```

Comment:
Why do we pass name twice? Can we assert `req.Name` with the response rather?

Reply:
@vnktram 
 here request is an interface requesttype will be  different for different resourcename and request may not always contain field name  in test case. Here request.Name means subscriptionName and name means resourceName

Reply:
@vnktram please suggest a way or should i keep it as it is?

Reply:
Added suggested changes for test case